yggdrasilffi/src/lib.rs

```diff
@@ -211,14 +211,21 @@ pub unsafe extern "C" fn check_variant(
     context_ptr: *const c_char,
     custom_strategy_results_ptr: *const c_char,
 ) -> *const c_char {
-    let result: Result<Option<VariantDef>, FFIError> = (|| {
+    let result: Result<Option<ExtendedVariantDef>, FFIError> = (|| {
         let engine = get_engine(engine_ptr)?;
         let toggle_name = get_str(toggle_name_ptr)?;
         let context: Context = get_json(context_ptr)?;
         let custom_strategy_results =
             get_json::<CustomStrategyResults>(custom_strategy_results_ptr)?;
-
-        Ok(engine.check_variant(toggle_name, &context, &Some(custom_strategy_results)))
+        let base_variant = engine.check_variant(
+            toggle_name,
+            &context,
+            &Some(custom_strategy_results.clone()),
+        );
+        let toggle_enabled = engine
```

Comment:
We'll be checking this twice since the consumer will have to anyway but the code that does the check is literally several orders of magnitude faster than moving across the FFI boundary so I think this is okay from a perf perspective